### PR TITLE
Mark files under `htmlcov/` as generated

### PR DESCRIPTION
**Pull request by spenserblack**
## Description
This marks files generated by [coverage.py](https://github.com/nedbat/coveragepy), specifically the `coverage html` command, as generated. There are currently [quite a lot](https://github.com/search?q=path%3A%2F%28%5E%7C%5C%2F%29htmlcov%5C%2F%2F&type=code) of these files that are returned by GitHub's search, and it seems like the large majority come from this tool ([more specific search results](https://github.com/search?q=path%3A%2F%28%5E%7C%5C%2F%29htmlcov%5C%2F.%2B_py%5C.html%24%2F&type=code)).

## Checklist:

- [x] **I am adding new or changing current functionality**
  - [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

**Comment by spenserblack** [comment]
I appear to be making some sort of mistake.
Tests fail, but running this locally seems to work fine (got the expected `appears to be generated source code` output for `htmlcov/index.html`).

**Comment by spenserblack** [comment]
Oh, of course. It helps to actually *read* the failure message. Moved the check up the chain of `||`s so that it's checked before any HTML, JS, or CSS is read.

Anyway, opening it back up for review. I'm anticipating that maybe some method renaming and/or combining with other methods may be requested :slightly_smiling_face:

**Review by Alhadis — CHANGES_REQUESTED** [review]
Minor nitpick concerning naming; ~~don't forget to move the method definition to its alphabetically-appropriate location in `generated.rb`~~ Scratch that, the file's not ordered anyway. 😅

File: lib/linguist/generated.rb [hunk 1]
```diff
@@ -72,6 +72,7 @@ def generated?
       generated_yarn_plugnplay? ||
       godeps? ||
       generated_by_zephir? ||
+      coveragepy_htmlcov_file? ||
```
[the remark this answers is not in the document]
Comment (spenserblack) [reply lib/linguist/generated.rb:75]: Good point, thanks!

**Review by Alhadis — APPROVED** [review]

**Pull request merged**